Core/include/Acts/Vertexing/SeedVertexFinder.hpp

```diff
@@ -0,0 +1,193 @@
+            const spacepoint_t& cc)
+        : a(aa),
+          b(bb),
+          c(cc),
+          ray(Acts::Vector3::Zero(), Acts::Vector3::Zero()) {}
+    const spacepoint_t &a, &b, &c;
+
+    Acts::Ray3D ray;
+  };
+
+  /// Configuration instance
+  Config m_cfg;
+
+  /// @brief Sorts spacepoints into a separate vectors for near, middle, and far spacepoints; and for each slice of phi
+  /// @param spacepoints Vector of the input spacepoints;
+  /// @return Vector of vectors for each set of spacepoints
+  std::vector<std::vector<std::vector<
+      std::vector<std::pair<spacepoint_t const*, Acts::ActsScalar>>>>>
+  sortSpacepoints(const std::vector<spacepoint_t>& spacepoints) const;
+
+  /// @brief
+  /// @param a
+  /// @param b
+  /// @return
+  static bool compareSpacepoints(
+      std::pair<spacepoint_t const*, Acts::ActsScalar>& a,
+      std::pair<spacepoint_t const*, Acts::ActsScalar>& b);
+
+  /// @brief Makes triplets from the provided vectors of near, middle, and far spacepoints; and for each slice of phi
+  /// @param sorted_spacepoints Vector of input vector of vectors for each set of spacepoints
+  /// @return Vector of valid triplets
+  std::vector<Triplet> findTriplets(
+      const std::vector<std::vector<std::vector<
+          std::vector<std::pair<spacepoint_t const*, Acts::ActsScalar>>>>>&
+          sorted_spacepoints);
+
+  /// @brief Validate the triplet based on "maxXYdeviation", "maxXYZdeviation", "maxZPosition", and "maxRPosition"
+  /// @param triplet A single triplet to be validated
+  /// @return True if the deviations and fitted ray are within the configured ranges
+  ///         If "minimalizeWRT"=="rays", then the fitted ray is also saved to
+  ///         the triplet for later
+  bool isTripletValid(Triplet& triplet);
+
+  /// @brief Calculates equation of the plane (alpha*x + beta*y + gamma*z + delta = 0), given the three points
+  /// @param triplet A single triplet (with 3 spacepoints)
+  /// @return A pair of {{alpha,beta,gamma},delta}
+  std::pair<Acts::Vector3, Acts::ActsScalar> makePlaneFromTriplet(
+      const Triplet triplet) const;
+
+  /// @brief Find a point (=the vertex) that has minimum chi^2 with respect to all planes defined by the triplets
+  /// @param triplets Vector of all valid triplets
+  /// @return Position {x,y,z} of the vertex
+  Acts::Vector3 findClosestPointFromPlanes(
+      const std::vector<Triplet>& triplets) const;
+
+  /// @brief Calculates parameters of the ray (starting point + direction), given the three points
+  /// @param triplet A single triplet (with 3 spacepoints)
+  /// @return A ray of {starting_point, direction}
+  Acts::Ray3D makeRayFromTriplet(const Triplet triplet) const;
+
+  /// @brief Find a point (=the vertex) that has minimum chi^2 with respect to all rays fitted through the triplets
+  /// @param triplets Vector of all valid triplets
+  /// @return Position {x,y,z} of the vertex
+  Acts::Vector3 findClosestPointFromRays(
```

Comment:
static?

Reply:
This depends on `m_cfg.minVtxShift` and `m_cfg.maxIterations`, so it can't be static. It's `const` though.